STANDARDS.md

```diff
@@ -1,16 +1,66 @@
-# Dataset Standards
 
-This file outlines the standards and requirements needed for submitting a dataset to QCArchive.
+This file outlines the standards and data requirements needed for submitting a dataset to QCArchive.
 This ensures that we have a consistent data model for downstream processes.
 
-## Required fields 
+STANDARDS version: 3-DRAFT (2020-11-13)
+
+We distinguish between standards for the datasets (i.e. the actual data), and the standards for training force fields.
 
-Current list:
-* Ensure all submissions have cmiles, most important are mapped hydrogen smiles
-* Ensure the WBO is requested for all submissions, this should be included in the scf properties list using the flag `wiberg_lowdin_indices`
+# Dataset Standards
 
-## Best practices
-* If any calculations are to be redone from another collection, re-use the old input (coordinates, atom ordering etc) as this will avoid running the calculation again and will just create new references in the database to the old results and should help keep the cost of the calculations down.  
+## Each molecule must have the following information:
+- Canonical isomeric explict hydrogen mapped SMILES
+- Provenence info of SMILES generation (NEW)
```

Comment:
I think this is definitely needed, but this will lead to a lot of repeated information as every molecule should have come through the same pipeline, so maybe it would be better to store this at a higher level like in the dataset metadata?

Reply:
So, we should make the distinction now I guess. Should the CMILES in the extras field be mutable?

Reply:
I am ok with this, but I think it could complicate things. For example, if I am a user and I see that provenance exists for one molecule but not another, is this a bug or expected behavior? What logic would I need to handle this? I would have to discover this in documentation somewhere that missing data means more top-level structures (e.g. entries) take precedence. My opinion here is that the data should be completely transparent (e.g. if it exists for one it should exist for all), or the actual QCA client provides a consistent interface and does the magic to find the right field if we have a sparser, more efficient representation. I am not a fan of having a sort of custom API layer outside of the client that people *must* use because we have enough tricks in the way we structure our data that it becomes unusable to non-experts.

One other issue is that a molecule can have multiple entries associated with it, e.g. it can belong in multiple datasets. In this case, if provenance is undefined for the molecule CMILES, which entry should I assume the provenance is from? We could apply (likely undocumented) logic here, but this could be avoided if we just applied provenance everywhere the function was used.